R/content.R

```diff
@@ -207,16 +207,18 @@ method(format, ContentToolRequest) <- function(
 ) {
   show <- arg_match(show)
 
-  if (length(x@arguments) == 0) {
-    call <- call2(x@name)
-  } else {
-    call <- call2(x@name, !!!x@arguments)
+  arguments <- tool_request_args(x)
+  if (is_tool_result(arguments)) {
```

Comment:
Maybe `tool_request_args()` should throw instead of creating the error result? It made sense when `tool_request_args()` was only used in the tool invocation flow; this feels a little out of place here.

I don't think it's a blocker for this PR; I'd be happy to follow up and refactor this later when I'm back